avocado/utils/nvme.py

```diff
@@ -0,0 +1,82 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# This code was inspired in the autotest project,
+#
+# client/base_utils.py
+#
+# Copyright: 2022 IBM
+# Authors : Naresh Bannoth <nbannoth@linux.vnet.ibm.com>
+
+
+"""
+Nvme utilities
+"""
+
+
+import os
+from avocado.utils import process
+
+
+class NvmeError(Exception):
+    """
+    Generic DiskError
+    """
+
+
+def get_controller_name(pci_addr):
+    """
+    Returns the controller/Adapter name with the help of pci_address
+
+    :param pci_addr: pci_address of the adapter
+
+    """
+    path = '/sys/bus/pci/devices/%s/nvme/' % pci_addr
+    return os.listdir(path)
+
+
+def get_number_of_ns_supported(controller_name):
+    """
+    Returns the number of namespaces supported for the nvme adapter
+
+    :param controller_name: Name of the controller eg: nvme0
+    :rtype: interger
+    """
+    cmd = 'nvme id-ctrl /dev/%s | grep ^nn' % controller_name
+    out = process.run(cmd, ignore_status=True, sudo=True,
+                      shell=True).stdout_text
+    return int(out.split(':')[-1].strip())
+
+
+def get_total_capacity(controller_name):
+    """
+    Returns the total capacity of the nvme adapter
+
+    :param controller_name: Name of the controller eg: nvme0
+    :rtype: interger
+    """
+    cmd = 'nvme id-ctrl /dev/%s | grep ^tnvmcap' % controller_name
```

Comment:
Possible to not use pipe?

Reply:
will explore more on this option

Reply:
yes this valid suggestion @Naresh-ibm please look on this